internal/services/queueService.go

```diff
@@ -56,31 +73,173 @@ func (qs *queueService) Listen() {
+			qs.logger.Infof("Processing handshake message")
+
+			supportedLanguages := languages.GetSupportedLanguagesWithVersions()
+			payload, err := json.Marshal(supportedLanguages)
+			if err != nil {
+				qs.logger.Errorf("Failed to marshal supported languages: %s", err)
+				err = PublishErrorToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, err)
+				if err != nil {
+					qs.logger.Errorf("Failed to publish error message: %s", err)
+				}
+				continue
+			}
+
+			err = PublishSucessToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, payload)
+			if err != nil {
+				qs.logger.Errorf("Failed to publish supported languages: %s", err)
+				err = PublishErrorToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, err)
+				if err != nil {
+					qs.logger.Errorf("Failed to publish error message: %s", err)
+				}
+			}
 		}
+	}
 
-		responseJSON, err := json.Marshal(response)
-		if err != nil {
-			qs.logger.Errorf("Failed to marshal response: %s", err)
-			continue
-		}
+}
 
-		qs.logger.Infof("Sending response: %s", response)
+func PublishErrorToResponseQueue(channel *amqp.Channel, responseQueueName, messageType, messageID string, err error) error {
 
-		err = qs.channel.Publish("", msg.ReplyTo, false, false, amqp.Publishing{
-			ContentType: "application/json",
-			Body:        responseJSON,
-		})
-		if err != nil {
-			qs.logger.Errorf("Failed to publish response: %s", err)
-		}
+	errorPayload := map[string]string{"error": err.Error()}
+	payload, jsonErr := json.Marshal(errorPayload)
+	if jsonErr != nil {
+		return jsonErr
+	}
+
+	queueMessage := QueueMessage{
+		Type:      messageType,
+		MessageID: messageID,
+		Payload:   payload,
+	}
+
+	responseJSON, jsonErr := json.Marshal(queueMessage)
+	if jsonErr != nil {
+		return jsonErr
+	}
+
+	err = channel.Publish("", responseQueueName, false, false, amqp.Publishing{
+		ContentType:   "application/json",
+		CorrelationId: messageID,
+		Body:          responseJSON,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func PublishSucessToResponseQueue(channel *amqp.Channel, responseQueueName, messageType, messageID string, payload []byte) error {
+	logger := logger.NewNamedLogger("queueService")
 
-		qs.logger.Infof("Response sent")
+	queueMessage := QueueMessage{
+		Type:      messageType,
+		MessageID: messageID,
+		Payload:   payload,
 	}
+
+	responseJSON, jsonErr := json.Marshal(queueMessage)
+	if jsonErr != nil {
+		return jsonErr
+	}
+
+	logger.Infof("Publishing response to queue: %s", responseQueueName)
+	err := channel.Publish("", responseQueueName, false, false, amqp.Publishing{
+		ContentType: "application/json",
+		Body:        responseJSON,
+	})
+
+	logger.Infof("Published response to queue: %s", responseQueueName)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (qs *queueService) RequeueTaskWithHigherPriority(queueMessage QueueMessage) error {
```

Comment:
Name of the function is different than what function do. I initialy assumed that it increases previos priority by 1. But instead it always sets 2. The name should be differnt IMHO